File: follow-the-beat/src/main/java/com/maco/followthebeat/v2/spotify/tracks/dto/SpotifyTrackDto.java
package com.maco.followthebeat.v2.spotify.tracks.dto;

import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
@Builder
public class SpotifyTrackDto {
    private String spotifyId;
    private String name;
    private Long durationMs;
    private Integer popularity;
    private String previewUrl;
    private boolean explicit;
    private AlbumDto album;
    private List<TrackArtistDto> artists;
}

File: follow-the-beat/react-beats/src/types/SpotifyTrackDto.ts
import SpotifyArtistDto from "./SpotifyArtistDto";

type SpotifyTrackDto = {
    spotifyId: string;
    name: string;
    durationMs: number;
    popularity: number;
    previewUrl: string;
    albumImgUrl: string;
    rank: number;
    artists: SpotifyArtistDto[];
}

export default SpotifyTrackDto;